src/Core/src/Handlers/WebView/WebViewHandler.Windows.cs

```diff
@@ -279,20 +273,19 @@ void InitialCookiePreloadIfNecessary(string url)
 
 			if (cookies is not null)
 			{
-				var existingCookies = GetCookiesFromPlatformStore(url);
-				foreach (HttpCookie cookie in existingCookies)
+				var existingCookies = await GetCookiesFromPlatformStore(url);
+				foreach (CoreWebView2Cookie cookie in existingCookies)
 				{
 					if (cookies[cookie.Name] is null)
-						myCookieJar.SetCookies(uri, cookie.ToString());
+						myCookieJar.SetCookies(uri,
+							$"{cookie.Name}={cookie.Value}; domain={cookie.Domain}; path={cookie.Path}");
```

Comment:
Having to deal with the value of a cookie is annoying. The less we touch the better, what about using the [following method to convert to a system.net.cookie](https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2cookie.tosystemnetcookie?view=webview2-dotnet-1.0.1587.40#microsoft-web-webview2-core-corewebview2cookie-tosystemnetcookie) and leave the set cookies method call as it was:
```suggestion
						myCookieJar.SetCookies(uri, cookie.ToSystemNetCookie ().ToString ());
```

Reply:
Good find! Oddly enough, I can't seem to find this method in our codebase... Are we using an older version of WebView2 where this is not yet available...?

Reply:
Changed it now to map all properties that overlap to a `System.Net.Cookie` and call `ToString()` on that and added a TODO to revisit later on the `ToSystemNetCookie`. Since I don't think it's crucial I guess a TODO is fine for now

Reply:
See also comment in another place here, we're using a different version of `CoreWebView2Cookie` which doesn't have this method: https://learn.microsoft.com/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2cookie?view=webview2-winrt-1.0.1587.40#expires